core/src/main/java/com/sequenceiq/cloudbreak/structuredevent/rest/urlparsers/RestUrlParser.java

```diff
@@ -0,0 +1,86 @@
+package com.sequenceiq.cloudbreak.structuredevent.rest.urlparsers;
+
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.ws.rs.container.ContainerRequestContext;
+
+public abstract class RestUrlParser {
+
+    public static final String RESOURCE_TYPE = "RESOURCE_TYPE";
+
+    public static final String RESOURCE_ID = "RESOURCE_ID";
+
+    public static final String RESOURCE_NAME = "RESOURCE_NAME";
+
+    public static final String RESOURCE_EVENT = "RESOURCE_EVENT";
+
+    public static final String WORKSPACE_ID = "WORKSPACE_ID";
+
+    public String getUrl(ContainerRequestContext requestContext) {
+        return requestContext.getUriInfo().getPath();
+    }
+
+    private String getMethod(ContainerRequestContext requestContext) {
+        return requestContext.getMethod();
+    }
+
+    public boolean fillParams(ContainerRequestContext requestContext, Map<String, String> params) {
+        if (parsedMethods().stream().anyMatch(method -> method.equals(getMethod(requestContext)))) {
+            return fillParams(getUrl(requestContext), params);
+        } else {
+            return false;
+        }
+    }
+
+    private boolean fillParams(String url, Map<String, String> params) {
+        if (checkAntiPattern(url)) {
+            return false;
+        }
+
+        Pattern pattern = Pattern.compile(getPattern());
+        Matcher matcher = pattern.matcher(url);
```

Comment:
getPattern() should return with the Pattern itself, Pattern would be a member variable of the subclass.

Reply:
fixed